watertap/unit_models/ion_exchange_0D.py

```
@@ -0,0 +1,1352 @@
+    CONFIG = ConfigBlock()
+
+    CONFIG.declare(
+        "dynamic",
+        ConfigValue(
+            domain=In([False]),
+            default=False,
+            description="Dynamic model flag - must be False",
+            doc="""Indicates whether this model will be dynamic or not,
+    **default** = False. The filtration unit does not support dynamic
+    behavior, thus this must be False.""",
+        ),
+    )
+
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(
+            default=False,
+            domain=In([False]),
+            description="Holdup construction flag - must be False",
+            doc="""Indicates whether holdup terms should be constructed or not.
+    **default** - False. The filtration unit does not have defined volume, thus
+    this must be False.""",
+        ),
+    )
+
+    CONFIG.declare(
+        "property_package",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+
+    CONFIG.declare(
+        "property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+
+    CONFIG.declare(
+        "target_ion",
+        ConfigValue(default="Ca_2+", domain=str, description="Target Ion"),
+    )
```

Are you using this to enforce a matching name between the property package and the unit model? If so, I don't think it makes sense to put in a default value. This would need to be something that is always user supplied and if the user does not supply anything, then it should report an error. 

Also, I am not certain that you would want to make this an `Enum`. That may limit the number of ions that a user could target. If there really only are a few possible targets, then maybe. But in general, you should be able to target any ion that exists in the property package.

---

Yes I would like it to be possible for any component to be the ``target_ion`` (or maybe ``target_component`` would be better). The idea is that the resin has high selectivity for the target and all other components would pass through.